### PR TITLE
Fix various issues in MapFission

### PR DESCRIPTION
**Pull request by tbennun**
Adds tests and fixes various issues in MapFission with multi-state nested SDFGs and complex subgraphs.

### CODE REVIEW

File: dace/sdfg.py [hunk 1]
```diff
@@ -4433,8 +4433,15 @@ def replsym(symlist):
             if isinstance(propclass, properties.RangeProperty):
                 setattr(node, pname, replsym(propval))
             if isinstance(propclass, properties.CodeProperty):
-                for stmt in propval['code_or_block']:
-                    ASTFindReplace({name: new_name}).visit(stmt)
+                if isinstance(propval['code_or_block'], str):
+                    # TODO: C++ AST parsing for replacement?
+                    if name != str(new_name):
+                        warnings.warn(
```
Comment (definelicht) [inline dace/sdfg.py]: Are we sure we don't want to fail here? Won't this break in most cases?

File: dace/transformation/helpers.py [hunk 2]
```diff
@@ -230,3 +230,45 @@ def nest_state_subgraph(sdfg: SDFG,
...
+                     or internal_memlet.subset[0] == 0)):
+            to_unsqueeze = ones[1:]
+        else:
+            to_unsqueeze = ones
+
+        result.subset.unsqueeze(to_unsqueeze)
+    elif len(internal_memlet.subset) > len(external_memlet.subset):
+        raise ValueError(
+            'Unexpected extra dimensions in internal memlet '
+            'while inlining SDFG.\nExternal memlet: %s\n'
```
Comment (definelicht) [inline dace/transformation/helpers.py]: "while inlining SDFG"?

File: dace/sdfg.py (hunk 1 above)
Comment (tbennun) [reply dace/sdfg.py]: It will only break in cases where you have symbols (e.g. `a[i] = i`) in the tasklet itself. In most cases it doesn't break.

Comment (definelicht) [reply dace/sdfg.py]: Ok, sure

File: dace/transformation/helpers.py (hunk 2 above)
Comment (tbennun) [reply dace/transformation/helpers.py]: Fixed, thanks

**Pull request merged**